src/services/rocketChat/index.js

```diff
@@ -482,6 +486,41 @@ class RocketChatChannel {
 		return Promise.resolve();
 	}
 
+	async synchronizeModerators(team) {
+		try {
+			const channel = await this.app.service('/rocketChat/channel').get(team._id);
+			const rcResponse = await request(getRequestOptions(
+				`/api/v1/groups.moderators?roomName=${channel.channelName}`,
+				{},
+				true,
+				undefined,
+				'GET',
+			));
+			let rcModerators = rcResponse.moderators;
+			const scModeratorPromises = [];
+			team.userIds.forEach(async (user) => {
+				if (this.teamModeratorRoles.includes(user.role.toString())) {
+					scModeratorPromises.push(this.app.service('rocketChat/user').get(user.userId));
+				}
+			});
+			let scModerators = await Promise.all(scModeratorPromises);
+			rcModerators = rcModerators.map(mod => mod._id);
+			scModerators = scModerators.map(mod => mod.rcId);
+			const addModerators = scModerators.filter(x => !rcModerators.includes(x));
+			const removeModerators = rcModerators.filter(x => !scModerators.includes(x));
+			addModerators.forEach(x => request(getRequestOptions(
+				'/api/v1/groups.addModerator', { roomName: channel.channelName, userId: x }, true,
+			)));
+			removeModerators.forEach(x => request(getRequestOptions(
+				'/api/v1/groups.removeModerator', { roomName: channel.channelName, userId: x }, true,
+			)));
+			return Promise.resolve();
+		} catch (err) {
+			logger.log(`Fehler beim Synchronisieren der rocket.chat moderatoren für team ${team._id} `, err);
+			return Promise.reject(err);
```

Comment:
promise reject could be placed after catch clause

Reply:
not neccessary in my opinion, as discussed in person